pkg/incentives/incentives.go

```diff
@@ -0,0 +1,371 @@
+// Copyright 2022 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package incentives
+
+import (
+	"context"
+	"crypto/rand"
+	"errors"
+	"fmt"
+	"io"
+	"math"
+	"sync"
+	"time"
+
+	"github.com/ethersphere/bee/pkg/log"
+	"github.com/ethersphere/bee/pkg/postage"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+type ChainBackend interface {
+	BlockNumber(context.Context) (uint64, error)
+}
+
+type Sampler interface {
+	ReserveSample(context.Context, []byte, uint8) ([]byte, error)
+}
+
+type Monitor interface {
+	IsStable() bool
+}
+
+var ErrSlashed = errors.New("slashed")
+
+type IncentivesContract interface {
+	ReserveSalt(context.Context) ([]byte, error)
+	IsPlaying(context.Context, uint8) (bool, error)
+	IsWinner(context.Context) (bool, bool, error)
+	Claim(context.Context) error
+	Commit(context.Context, []byte) error
+	Reveal(context.Context, uint8, []byte, []byte) error
+	WrapCommit(uint8, []byte, []byte, []byte) ([]byte, error)
+}
+
+const loggerName = "incentives"
```

Comment:
That should be first after the imports with comments on it as it is done in the other packages.